presto-bytecode/src/main/java/com/facebook/presto/bytecode/MethodDefinition.java

```diff
@@ -84,10 +84,9 @@ public MethodDefinition(
         this.parameters = ImmutableList.copyOf(parameters);
         this.parameterTypes = Lists.transform(this.parameters, Parameter::getType);
         this.parameterAnnotations = ImmutableList.copyOf(transform(parameters, input -> new ArrayList<>()));
-
         Optional<ParameterizedType> thisType = Optional.empty();
-        if (!access.contains(STATIC)) {
-            thisType = Optional.of(declaringClass.getType());
+        if (!declaringClass.isInterface() && !access.contains(STATIC)) {
```

Comment:
add validation for these illegal combinations to the `ClassDefinition` constructor